structuredheadings/plugin.js

```diff
@@ -294,41 +294,47 @@
           this.add("restart", "Restart Styling", "Restart Styling");
         },
 
+        clearAutonumberClassesForHeading: function (heading) {
+          clearStyles(editor, heading);
+          clearLevel(editor, heading);
+          clearNumbering(editor, heading);
+        },
+
+        setAutonumberClassesForHeading: function (value, heading) {
+          setNumbering(editor, heading);
+          setLevel(editor, heading);
+          setCurrentStyle(editor, heading, value);
+        },
+
         onClick: function (value) {
-          var block = CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor);
           if (value === "restart") {
             editor.execCommand("restartNumbering");
-          } else if (value === "clear") {
-            clearStyles(editor, block);
-            clearLevel(editor, block);
-            clearNumbering(editor, block);
-          } else {
-            // convert any non-numbered headings to numbered
-            var selection = editor.getSelection();
-            if (selection.getRanges()[0].collapsed) {
-              this.setAutonumberClassesForHeading(block, value);
-            } else {
-              var headings = CKEDITOR.plugins.structuredheadings.getHeadingsInSelection(
+            return;
+          }
+          var selection = editor.getSelection();
+          var headings = CKEDITOR.plugins.structuredheadings.getHeadingsInSelection(
                 editor,
                 selection
               );
-              if (headings.length > 0) {
-                for (var i = 0; i < headings.length; i++) {
-                  this.setAutonumberClassesForHeading(headings[i], value);
-                }
-              }
+          var func;
+
+          if (value === "clear") {
+            func = this.clearAutonumberClassesForHeading;
+          } else {
+            func = this.setAutonumberClassesForHeading.bind(this, value);
+          }
+
+          if (headings.length > 0) {
+            for (var i = 0; i < headings.length; i++) {
+              func(headings[i]);
             }
-            // apply the correct bulletstyle for all numbered headings
-            editor.execCommand("reapplyStyle", value);
-            // set the combo box value
-            this.setValue(value, value);
           }
-        },
 
-        setAutonumberClassesForHeading: function (heading, value) {
-          setNumbering(editor, heading);
-          setLevel(editor, heading);
-          setCurrentStyle(editor, heading, value);
+          // apply the correct bulletstyle for all numbered headings
+          editor.execCommand("reapplyStyle", value);
```

Comment:
It is indirectly tested.  It's some of the leftover code i haven't touched yet.

It goes through all the autonumbered headings and basically ensures that the currently selected bullet type is re-applied to all of them (you can only have one autonumbering type at a time currently.).

So any of the multi-heading related tests indirectly test it